### ISSUE TITLE
Weird scrolling behavior

### ISSUE THREAD

**Issue by DUOLabs333** [utmapp/UTM#5693]
**BEFORE SUBMITTING YOUR ISSUE, PLEASE LOOK AT THE PINNED ISSUES AND USE THE SEARCH FUNCTION TO MAKE SURE IT IS NOT ALREADY REPORTED. ALWAYS COMMENT ON AN EXISTING ISSUE INSTEAD OF MAKING A NEW ONE.**

**Describe the issue**  
When scrolling with an external mouse slowly, you may have to turn the wheel 2-4 times before the scrollbar reacts. This happens on every app I tried, both GUI and TUI. This does not happen with the trackpad --- you can drag slowly and the scrollbar will still react.

**Configuration**  
* UTM Version: 4.4.0
* macOS Version: 11
* Mac Chip (Intel, M1, ...): M1

**Crash log**  

**Debug log**  

**Upload VM**

**Comment by osy** [comment]
Yeah this is a side effect of how SPICE sends the scroll events. It translates them to button presses (up or down) and sends the button presses to QEMU which gets sent to the guest and the guest re-intreprets them as scroll. My guess is that in this process, the guest does not see the scroll until a few "presses" in a single direction.

**Comment by osy** [comment]
I think a proper fix would be to pass the scroll events directly to QEMU as HID events. Or a new SPICE channel to send the scroll events directly.

**Comment by alberthastings** [comment]
I also suffer from this problem.

Processor: Intel Core i5
O.S.: MacOS Ventura 13.6.1
UTM version: 4.4.4 (92)
Mode: Virtualized and with QEmu Engine
Guest O.S.: Debian 12.2.0 x64

I just recently migrated from VirtualBox.
It really strikes me that, until now (2 months ago), no one has reported this REAL problem.
Please fix it as soon as possible because, in this case, the use of UTM becomes practically unfeasible.

**Comment by Robpeter** [comment]
Any workarounds on this mouse issue?

**Comment by sokurenko** [comment]
I was forced to buy magic trackpad, it works excellent.

**Comment by kmatheussen** [comment]
Last comment in the other ticket has a working workaround: https://github.com/utmapp/UTM/issues/3417#issuecomment-2139633302

In the Mos program (which is the fix), I had to set "Step", "Speed", and "Duration" to minimum values though, and it still scrolls more than I would like, but at least it scrolls immediately now.

**Comment by kmatheussen** [comment]
Hold on, "Duration" has to be set to maximum value, not minimum value. But now the mouse wheel scrolls far too slow in the host. Well, I guess it's possible to configure both host and client to be decent somehow.

**Comment by alexpyattaev** [comment]
If you own a fancy mouse you can set app-specific scroll speed in your mouse software's settings. Or alternatively brew install mos and cross the fingers that it does not steal your passwords.

**Comment by DUOLabs333** [comment]
I decided to look into this today --- I noticed that even on macOS, scrolling the mouse wheel leads to inconsistent scrolling in windows (I used Firefox and Chrome in my experiments) --- it's almost like there is some resistance I didn't notice before.

**Comment by alexpyattaev** [comment]
What do you mean by resistance? How to test if it is present?

**Comment by DUOLabs333** [comment]
@alexpyattaev To see if you experience any resistance, on the host OS, see if every scroll of the mouse wheel corresponds to a movement of the scrollbar up and down. If there is some "resistance", you should see that some scrolls up do not move the scrollbar up by the same amount, and the same thing with scrolls down.

**Comment by alexpyattaev** [comment]
I have not manage to observe this in any of the native apps.

**Comment by ushabuntu** [comment]
Issue still active - UTM 4.6.4 | M1 Macbook Air | Sequoia 15.5

**Comment by doganov** [comment]
I am also hitting the same issue - UTM 4.7.5 | M4 MacBook Air | Sequoia 15.7.4 as host | Debian trixie (arm64) as guest. Certain kinds of work are impractical on the guest because of this.
